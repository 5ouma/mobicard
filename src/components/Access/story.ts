import type { Props } from "./Access.astro";

export const Default: Props = {
  homepage: "https://example.com",
  email: "email@example.com",
};

export const SubDomain: Props = {
  homepage: "https://www.example.com",
  email: "email@www.example.com",
};

export const WithoutEmail: Props = {
  homepage: "https://www.example.com",
};
