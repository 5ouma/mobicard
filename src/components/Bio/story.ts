import type { Props } from "./Bio.astro";

export const Default: Props = {
  name: "Bio Name",
  description: "Bio Description",
  icon: "https://github.com/withastro.png",
};

export const Japanese: Props = {
  name: "名前",
  description: "説明",
  icon: "https://github.com/withastro.png",
};

export const Long: Props = {
  name: "Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Name",
  description:
    "Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Long Bio Description",
  icon: "https://github.com/withastro.png",
};
