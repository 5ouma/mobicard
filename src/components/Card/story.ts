import type { Props } from "./Card.astro";

export const Default: Props = { url: "https://example.com" };
export const SubDomain: Props = { url: "https://www.example.com" };
