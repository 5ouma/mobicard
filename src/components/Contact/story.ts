import type { Props } from "./Contact.astro";

export const Bluesky: Props = {
  service: "Bluesky",
  id: "bsky.app",
};

export const Facebook: Props = {
  service: "Facebook",
  id: "facebook",
};

export const GitHub: Props = {
  service: "GitHub",
  id: "github",
};

export const Instagram: Props = {
  service: "Instagram",
  id: "instagram",
};

export const Mastodon: Props = {
  service: "Mastodon",
  id: "@Mastodon@mastodon.social",
};

export const Misskey: Props = {
  service: "Misskey",
  id: "@support@misskey.io",
};

export const Posts: Props = {
  service: "Posts",
  id: "support",
};

export const Twitter: Props = {
  service: "Twitter",
  id: "twitter",
};

export const Mail: Props = {
  service: "Mail",
  id: "mail@example.com",
};
