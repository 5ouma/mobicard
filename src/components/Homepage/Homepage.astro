---
import Svg from "@cxa/astro-inline-svg";
import { Icon } from "astro-icon/components";

import "../../styles/global.css";
import { generateQRCode } from "../../libs/qrcode.ts";

export type Props = { url: string };
const props = Astro.props;

const url = new URL(props.url);
const qrCode: string = await generateQRCode(props.url);
---

<a
  href={props.url}
  target="_blank"
  rel="noopener noreferrer"
  aria-label={`Visit ${url.hostname}`}
>
  <Svg id="qr" raw={qrCode} />
  <div>
    <Icon id="icon" name="mingcute:home-3-line" />
    <span aria-hidden="true">
      {`${url.hostname}${url.pathname === "/" ? "" : url.pathname}`}
    </span>
  </div>
</a>

<style>
  a {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 0.3rem;
    text-decoration: none;
    &:hover div {
      border-color: color-mix(in srgb, var(--foreground) 60%, transparent);
    }
  }

  #qr {
    path {
      stroke: var(--foreground);
    }
    width: 6rem;
    height: auto;
  }

  div {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.3rem;
    padding: 0 0.3rem;
    color: var(--foreground);
    border-bottom: 2px solid
      color-mix(in srgb, var(--foreground) 30%, transparent);
    transition: all 0.3s ease-in-out;
  }

  #icon {
    width: 1rem;
    height: auto;
  }

  span {
    font-family: "LINE Seed JP", sans-serif;
    font-size: 0.75rem;
    font-style: normal;
    font-weight: normal;
    line-height: normal;
  }
</style>
