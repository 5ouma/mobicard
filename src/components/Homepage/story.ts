import type { Props } from "./Homepage.astro";

export const Domain: Props = { url: "https://example.com" };
export const Path: Props = { url: "https://www.example.com/path" };
export const Japanese: Props = { url: "https://www.example.com/パス" };
