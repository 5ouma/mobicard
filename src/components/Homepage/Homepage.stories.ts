import Homepage from "./Homepage.astro";
import * as stories from "./story.ts";

export default {
  title: "Homepage",
  component: Homepage,
};

export const Domain = { args: stories.Domain };
export const Path = { args: stories.Path };
export const Japanese = { args: stories.Japanese };
