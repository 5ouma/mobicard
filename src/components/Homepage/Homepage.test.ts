import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { describe, expect, test } from "vitest";

import Homepage from "./Homepage.astro";
import type { Props } from "./Homepage.astro";
import * as stories from "./story.ts";

describe("Homepage", () => {
  describe("Valid URL", () => {
    for (const [name, props] of Object.entries(stories)) {
      test(name, async () => {
        const container: AstroContainer = await AstroContainer.create();
        const result: string = await container.renderToString(Homepage, {
          props,
        });

        expect(result).toMatchSnapshot();
      });
    }
  });

  test("Invalid URL", async () => {
    const container: AstroContainer = await AstroContainer.create();

    await expect(
      container.renderToString(Homepage, {
        props: { url: "not-valid-url" } satisfies Props,
      }),
    ).rejects.toThrow();
  });
});
