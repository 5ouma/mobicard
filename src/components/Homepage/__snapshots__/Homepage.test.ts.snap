// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > Valid URL > Domain 1`] = `"<a href="https://example.com" target="_blank" rel="noopener noreferrer" aria-label="Visit example.com" data-astro-cid-q5ccfyou> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 25 25" shape-rendering="crispEdges" id="qr" data-astro-cid-q5ccfyou><path stroke="#000000" d="M0 0.5h7m4 0h3m2 0h1m1 0h7M0 1.5h1m5 0h1m3 0h1m2 0h4m1 0h1m5 0h1M0 2.5h1m1 0h3m1 0h1m1 0h2m1 0h1m2 0h1m3 0h1m1 0h3m1 0h1M0 3.5h1m1 0h3m1 0h1m1 0h1m4 0h3m2 0h1m1 0h3m1 0h1M0 4.5h1m1 0h3m1 0h1m1 0h3m2 0h1m2 0h1m1 0h1m1 0h3m1 0h1M0 5.5h1m5 0h1m1 0h1m2 0h1m2 0h2m2 0h1m5 0h1M0 6.5h7m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h7M8 7.5h1m5 0h1m1 0h1M0 8.5h1m1 0h5m5 0h1m5 0h5M1 9.5h1m2 0h2m2 0h1m1 0h2m1 0h1m3 0h1m1 0h1m3 0h1M0 10.5h5m1 0h1m1 0h2m3 0h4m2 0h1m1 0h1m1 0h2M0 11.5h2m1 0h3m2 0h1m1 0h2m1 0h1m1 0h2m1 0h2m4 0h1M1 12.5h3m2 0h1m4 0h2m1 0h2m1 0h2m1 0h1m1 0h3M0 13.5h5m3 0h1m1 0h1m5 0h1m2 0h1m1 0h1m1 0h1M0 14.5h1m5 0h2m2 0h3m2 0h1m2 0h4m1 0h2M0 15.5h1m2 0h1m3 0h1m3 0h1m2 0h7m3 0h1M0 16.5h1m1 0h1m2 0h2m1 0h4m4 0h5m1 0h1M8 17.5h2m2 0h5m3 0h2M0 18.5h7m6 0h2m1 0h1m1 0h1m1 0h1m1 0h3M0 19.5h1m5 0h1m1 0h2m2 0h2m2 0h1m3 0h2m1 0h1M0 20.5h1m1 0h3m1 0h1m1 0h3m1 0h1m1 0h7m1 0h1m1 0h1M0 21.5h1m1 0h3m1 0h1m1 0h1m6 0h1m1 0h2m1 0h5M0 22.5h1m1 0h3m1 0h1m1 0h5m2 0h1m5 0h2m1 0h1M0 23.5h1m5 0h1m4 0h1m2 0h1m1 0h2m1 0h3m2 0h1M0 24.5h7m1 0h2m1 0h1m5 0h8"></path></svg> <div data-astro-cid-q5ccfyou> <svg width="1em" height="1em" id="icon" data-astro-cid-q5ccfyou data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-q5ccfyou> example.com </span> </div> </a> "`;

exports[`Homepage > Valid URL > Japanese 1`] = `"<a href="https://www.example.com/パス" target="_blank" rel="noopener noreferrer" aria-label="Visit www.example.com" data-astro-cid-q5ccfyou> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 29 29" shape-rendering="crispEdges" id="qr" data-astro-cid-q5ccfyou><path stroke="#000000" d="M0 0.5h7m1 0h1m8 0h2m3 0h7M0 1.5h1m5 0h1m3 0h1m5 0h1m2 0h1m2 0h1m5 0h1M0 2.5h1m1 0h3m1 0h1m2 0h5m1 0h6m1 0h1m1 0h3m1 0h1M0 3.5h1m1 0h3m1 0h1m1 0h1m2 0h2m1 0h3m2 0h1m2 0h1m1 0h3m1 0h1M0 4.5h1m1 0h3m1 0h1m1 0h2m2 0h3m1 0h3m3 0h1m1 0h3m1 0h1M0 5.5h1m5 0h1m1 0h3m2 0h3m1 0h1m1 0h1m2 0h1m5 0h1M0 6.5h7m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h7M8 7.5h1m1 0h2m2 0h1m3 0h2M0 8.5h1m3 0h1m1 0h3m2 0h1m1 0h5m1 0h7m2 0h1M0 9.5h2m1 0h2m2 0h1m1 0h2m7 0h1m3 0h7M1 10.5h3m1 0h4m4 0h3m6 0h3m3 0h1M4 11.5h2m1 0h1m1 0h2m1 0h2m1 0h2m1 0h3m1 0h1m1 0h2m1 0h2M2 12.5h1m2 0h2m1 0h2m1 0h3m4 0h1m1 0h2m5 0h1M1 13.5h3m1 0h1m1 0h2m1 0h1m2 0h2m1 0h1m1 0h1m1 0h3m1 0h5M2 14.5h2m2 0h2m2 0h1m1 0h1m3 0h1m3 0h2m2 0h3m1 0h1M2 15.5h1m4 0h2m2 0h2m1 0h1m3 0h1m1 0h3m4 0h2M1 16.5h1m1 0h4m1 0h2m1 0h6m1 0h4m1 0h1m3 0h1M0 17.5h3m1 0h1m2 0h3m2 0h1m9 0h4m1 0h2M2 18.5h1m1 0h1m1 0h2m1 0h1m1 0h1m1 0h3m1 0h1m3 0h2m1 0h1m1 0h1m1 0h1M2 19.5h1m1 0h2m1 0h1m1 0h1m1 0h1m1 0h1m1 0h2m1 0h1m1 0h1m1 0h1m4 0h2M0 20.5h2m1 0h4m1 0h1m1 0h4m3 0h1m2 0h6m2 0h1M8 21.5h1m3 0h3m1 0h1m1 0h3m3 0h1m3 0h1M0 22.5h7m1 0h1m7 0h1m2 0h2m1 0h1m1 0h3m1 0h1M0 23.5h1m5 0h1m4 0h2m1 0h1m3 0h3m3 0h1m2 0h1M0 24.5h1m1 0h3m1 0h1m1 0h4m1 0h3m2 0h8m1 0h2M0 25.5h1m1 0h3m1 0h1m2 0h2m1 0h2m4 0h4m1 0h1m3 0h1M0 26.5h1m1 0h3m1 0h1m2 0h1m1 0h2m1 0h2m5 0h1m3 0h4M0 27.5h1m5 0h1m2 0h2m2 0h4m2 0h2m3 0h2m1 0h2M0 28.5h7m1 0h1m2 0h3m1 0h4m1 0h2m1 0h2m2 0h1"></path></svg> <div data-astro-cid-q5ccfyou> <svg width="1em" height="1em" id="icon" data-astro-cid-q5ccfyou data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-q5ccfyou> www.example.com/%E3%83%91%E3%82%B9 </span> </div> </a> "`;

exports[`Homepage > Valid URL > Path 1`] = `"<a href="https://www.example.com/path" target="_blank" rel="noopener noreferrer" aria-label="Visit www.example.com" data-astro-cid-q5ccfyou> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 29 29" shape-rendering="crispEdges" id="qr" data-astro-cid-q5ccfyou><path stroke="#000000" d="M0 0.5h7m1 0h3m1 0h4m1 0h4m1 0h7M0 1.5h1m5 0h1m1 0h2m3 0h1m2 0h3m3 0h1m5 0h1M0 2.5h1m1 0h3m1 0h1m4 0h1m4 0h1m1 0h1m1 0h1m1 0h1m1 0h3m1 0h1M0 3.5h1m1 0h3m1 0h1m1 0h2m2 0h1m1 0h2m1 0h3m2 0h1m1 0h3m1 0h1M0 4.5h1m1 0h3m1 0h1m7 0h1m1 0h1m2 0h1m2 0h1m1 0h3m1 0h1M0 5.5h1m5 0h1m4 0h1m1 0h1m1 0h6m1 0h1m5 0h1M0 6.5h7m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h7M8 7.5h2m2 0h1m2 0h6M0 8.5h1m1 0h2m1 0h3m4 0h2m1 0h1m1 0h3m2 0h1m2 0h1m1 0h2M0 9.5h1m2 0h2m2 0h1m2 0h6m3 0h6m3 0h1M1 10.5h3m1 0h6m2 0h1m2 0h2m2 0h2m1 0h2m1 0h2M1 11.5h1m1 0h2m3 0h5m3 0h1m3 0h5m3 0h1M0 12.5h1m1 0h6m4 0h1m1 0h2m1 0h1m1 0h1m5 0h2M1 13.5h3m1 0h1m3 0h1m2 0h1m1 0h1m1 0h2m1 0h5m2 0h3M1 14.5h2m2 0h2m5 0h2m1 0h2m2 0h2m2 0h2m1 0h3M0 15.5h3m1 0h1m2 0h3m1 0h1m2 0h1m1 0h1m3 0h2m1 0h2m2 0h1M0 16.5h1m1 0h2m2 0h1m1 0h2m1 0h1m2 0h3m1 0h1m1 0h2m2 0h2m1 0h1M2 17.5h1m2 0h1m1 0h4m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m2 0h1m1 0h3M0 18.5h1m1 0h1m1 0h3m1 0h1m2 0h1m1 0h4m4 0h1m1 0h1m2 0h1M3 19.5h3m1 0h1m1 0h1m3 0h2m2 0h1m3 0h1m4 0h1M1 20.5h1m1 0h2m1 0h1m2 0h2m1 0h3m1 0h3m1 0h7M8 21.5h1m1 0h2m3 0h4m1 0h1m3 0h5M0 22.5h7m1 0h4m2 0h2m1 0h1m1 0h2m1 0h1m1 0h2m1 0h1M0 23.5h1m5 0h1m1 0h1m2 0h1m1 0h3m4 0h1m3 0h2m2 0h1M0 24.5h1m1 0h3m1 0h1m2 0h3m4 0h1m1 0h1m1 0h5m1 0h1m1 0h1M0 25.5h1m1 0h3m1 0h1m1 0h2m1 0h1m8 0h2m2 0h2m1 0h1M0 26.5h1m1 0h3m1 0h1m1 0h4m2 0h1m2 0h3m3 0h1m2 0h1m1 0h1M0 27.5h1m5 0h1m3 0h1m3 0h2m1 0h4m1 0h2m1 0h1m1 0h1M0 28.5h7m1 0h2m1 0h2m4 0h5m3 0h1m1 0h1"></path></svg> <div data-astro-cid-q5ccfyou> <svg width="1em" height="1em" id="icon" data-astro-cid-q5ccfyou data-icon="mingcute:home-3-line">   <symbol id="ai:mingcute:home-3-line" viewBox="0 0 24 24"><g fill="none"><path d="m12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035q-.016-.005-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427q-.004-.016-.017-.018m.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093q.019.005.029-.008l.004-.014l-.034-.614q-.005-.018-.02-.022m-.715.002a.02.02 0 0 0-.027.006l-.006.014l-.034.614q.001.018.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01z"/><path fill="currentColor" d="M10.772 2.688a2 2 0 0 1 2.317-.099l.139.1l8.384 6.52c.721.561.37 1.692-.499 1.785l-.116.006H20v8a2 2 0 0 1-1.85 1.995L18 21H6a2 2 0 0 1-1.994-1.85L4 19v-8h-.997C2.09 11 1.671 9.892 2.3 9.285l.088-.076zM12 4.267L5.625 9.225c.229.185.375.468.375.785V19h3v-5a3 3 0 1 1 6 0v5h3v-8.99c0-.317.146-.6.375-.785zM12 13a1 1 0 0 0-1 1v5h2v-5a1 1 0 0 0-1-1"/></g></symbol><use href="#ai:mingcute:home-3-line"></use>  </svg> <span aria-hidden="true" data-astro-cid-q5ccfyou> www.example.com/path </span> </div> </a> "`;
